iib/workers/api_utils.py

```diff
@@ -168,6 +168,14 @@ def update_request(
             rv.status_code,
             rv.text,
         )
+        rv_error = rv.json().get('error', None)
+        if rv_error in [
+            "A failed request cannot change states",
+            "A complete request cannot change states",
+        ]:
+            raise FinalStateAlreadyReached(
```

Comment:
Please correct me if I am mistaken, but it looks like the change is gracefully handling the second execution of the request and not preventing it from being executed again in the first place?

Reply:
While this PR will not solve the initially thought bug, but isn't it better to have this improvement as we already have it now?
With this change the time spent on retries for modifying the failed requests would be saved. 
IMO it will surely make sense to close the PR if its adding too much complication, but if its not and the benefits outweigh the complication, it could be worth keeping it.